pkg/tsdb/postgres/macros_test.go

```diff
@@ -319,6 +319,68 @@ func TestMacroEngine(t *testing.T) {
 		})
 	})
 
+	t.Run("Macro: $__constructPredicates", func(t *testing.T) {
+
+		t.Run("without arguments should return True", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, "$__constructPredicates()")
+
+			require.NoError(t, err)
+			require.Equal(t, "true", result)
+		})
+
+		t.Run("single param - NULL value should return true", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, `$__constructPredicates("var1:NULL")`)
+
+			require.NoError(t, err)
+			require.Equal(t, "true", result)
+		})
+
+		t.Run("single param - NULL value prepended with whitespaces should return true", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, `$__constructPredicates("var1:   NULL")`)
+
+			require.NoError(t, err)
+			require.Equal(t, "true", result)
+		})
+
+		t.Run("single param - should add var1 with corresponding value as the predicate", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, `$__constructPredicates("var1: 'abc'")`)
+
+			require.NoError(t, err)
+			require.Equal(t, `var1 in ('abc')`, result)
+		})
+
+		t.Run("single param - should add var1 with corresponding value as the predicate when value contains quotes", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, `$__constructPredicates("var1: '2/2" Z"Z'")`)
+
+			require.NoError(t, err)
+			require.Equal(t, `var1 in ('2/2" Z"Z')`, result)
+		})
+
+		t.Run("multiple values - should add single param with multiple values in the constructed predicate", func(t *testing.T) {
+			result, err := engine.Interpolate(query, backend.TimeRange{}, `$__constructPredicates("var1: '1/2" ZZ AAA','1/4' XX AAA','3/8'")`)
+
+			require.NoError(t, err)
+			require.Equal(t, `var1 in ('1/2" ZZ AAA','1/4' XX AAA','3/8')`, result)
```

Comment:
From a sql execution perspective, i feel this will create problem in the second parameter as the parameter in the single quote is closed in the middle. Within a param single quote must be escaped with another single quote

That is the second parameter should become `1/4'' XX AAA` in the above it is `1/4' XX AAA`. Notice the second single quote which excapes the single quote

Reply:
For science you can try this and see for yourself :-)

```sql
create table abc(var1 text);

select * from abc where var1 in ('1/2" ZZ AAA','1/4' XX AAA','3/8');

drop table abc;
```

Reply:
Have Modified the implementation to fix the above